pkg/controller/clusterpool/collections.go

```diff
@@ -913,18 +924,39 @@ func (cdcs *cdcCollection) SyncClusterDeploymentCustomizationAssignments(c clien
 		// CDC exists
 		cdc, ok := cdcs.namespace[cpRef.CustomizationRef.Name]
 		if !ok {
+			logger.Warningf("CD %s has CDC %s that doesn't exist, this is a bug", cd.Name, cpRef.CustomizationRef.Name)
 			continue
 		}
 
-		// CDC is no reserved
+		// CDC is not reserved
 		available := conditionsv1.FindStatusCondition(cdc.Status.Conditions, conditionsv1.ConditionAvailable)
-		if available == nil || available.Status == corev1.ConditionTrue {
+		if available == nil || available.Status == corev1.ConditionUnknown || available.Status == corev1.ConditionTrue {
+			// CDC is used by other CD
```

Comment:
Let's add some logging in this branch. I think an initial message saying we found a conflict in reservation status; and then another when we find out whether the other CD exists or not.